src/components/views/rooms/RoomTile.tsx

```diff
@@ -576,6 +597,67 @@ export default class RoomTile extends React.PureComponent<IProps, IState> {
         );
     }
 
+    private updateVoiceParticipants = (participants: IJitsiParticipant[]) => {
+        this.setState({ voiceParticipants: participants });
+    };
+
+    private renderVoiceChannel(): React.ReactElement {
+        if (!this.state.voiceParticipants.length) return null;
+
+        const faces = this.state.voiceParticipants.map(p =>
+            <BaseAvatar
+                key={p.participantId}
+                name={p.displayName ?? p.formattedDisplayName}
+                idName={p.participantId}
+                url={p.avatarURL}
+                width={24}
+                height={24}
+            />,
+        );
+
+        // TODO: The below "join" button will eventually show up on text rooms
+        // with an active voice channel, but that isn't implemented yet
+        return <div className="mx_RoomTile_voiceChannel">
+            <FacePile faces={faces} overflow={false} />
+            { this.isVoiceRoom ? null : (
+                <AccessibleButton
+                    kind="link"
+                    className="mx_RoomTile_connectVoiceButton"
+                    onClick={this.connectVoice.bind(this)}
+                >
+                    { _t("Join") }
+                </AccessibleButton>
+            ) }
+        </div>;
+    }
+
+    private async connectVoice() {
+        this.setState({ voiceConnectionState: VoiceConnectionState.Connecting });
+        // TODO: Actually wait for the widget to be ready, instead of guessing.
+        // This hack is only in place until we find out for sure whether design
+        // wants the room view to open when connecting voice, or if this should
+        // somehow connect in the background. Until then, it's not worth the
+        // effort to solve this properly.
+        await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
there's a widget messaging onReady hook somewhere, if useful when implemented

Reply:
Thanks, will keep an eye out for that later